homeserver_synapse.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
```

Comment:
Not a Go expert here; should these `homeserver_xxx.go` files be in a subdirectory rather than the repository root, to keep things a little bit tidier?

Reply:
That's what I'd do in another PR. :)
Some cleanup here and there.

Reply:
Sure, happy to take that to another PR